### PR TITLE
bootstrap: Whenever we error out we inform the user

### PR DESCRIPTION
**Pull request by LiliC**
As agreed we want to show the user when the bootstrap errors out in the UI as well, we do this by adding a new event `onboarding_failed`. In the next PR I will add to send errors whenever bootstrap itself fails or anything in the install script.

screen shot 2018-04-20 at 2 20 10 pm

After this is merged and before we ship to production, we need to add the event type `onboarding_failed` to production as well, as it's currently only on dev and local. Will do a PR for that as well.

### CODE REVIEW

File: bootstrap/main.go [hunk 1]
```diff
@@ -64,30 +68,39 @@ func mainImpl() {
...
 	wcOrgLookupURL, err := text.ResolveString(weavecloud.DefaultWCOrgLookupURLTemplate, opts)
 	if err != nil {
+		sendError("Error occurred during installing agents.", opts)
 		log.Fatal("invalid URL template:", err)
 	}
 
 	// Restore stdin, making fd 0 point at the terminal
 	if err := syscall.Dup2(1, 0); err != nil {
+		sendError("Could not restore stdin", opts)
 		exitWithCapture("Could not restore stdin\n", err)
```
Comment (rade) [inline bootstrap/main.go]: The repeated pairing of `sendError` and `exitWithCapture`, with near-identical-content, should be replaced with a function that does both. And I see no reason for the text to differ.

Comment (LiliC) [reply bootstrap/main.go]: My reasoning behind it, as we have in some places instructions that say `...run kubectl version....`, etc., and I think displaying that in the UI is not the right place but the CLI is. 

The content of the `sendError` msgs might change, as @ngehani might have different suggestions.

But if it's okay with everyone, I am happy to just call `sendError` in `exitWithCapture`.

Comment (rade) [reply bootstrap/main.go]: It's ok to say "run kubectl version" in the UI. As for the content of the error messages changing - I would expect any changes to affect both places. There's certainly a lot to be said for showing the same error in the UI as in the CLI.

Comment (LiliC) [reply bootstrap/main.go]: Ok, then will return the same error to CLI and UI, sgtm. 👍

**Comment by ngehani** [comment]
@LiliC - What is the guidance for the user when they see that error? Space is a problem if the explanation is too long. If there's way to provide what things the user should check/fix, it would be good.

**Comment by rade** [comment]
This PR is a (significant) net improvement over the status quo. Let's revise error messages in a separate issue (@LiliC please file one).

**Comment by dlespiau** [comment]
Oops, hadn't seen Marcus already approved!

**Pull request merged**